### PR TITLE
Expiration of Health Cards

### PR DESCRIPTION
**Pull request by pvillela**
Introducing the ability to expire Health Cards and the rationale for doing so in special situations.

### CODE REVIEW

File: docs/index.md
Comment (p2-apple) [inline docs/index.md:435]:
```suggestion
One use case to issuing Health Cards with an expiration date is a government entity issuing a vaccination Health Card to foreign visitors for their use while in the country. This visitor's vaccination card is issued based on original documents presented by the visitor. Even with robust verification protocols, the government entity may not want to vouch for the validity of the visitor pass for an unlimited period of time. Importantly, the original document may be invalidated at some point in the future, e.g. by its signing keys being revoked. It may be impractical for the government entity issuing the visitor pass to track and reactively revoke the visitor pass. This risk can be mitigated by setting an expiration date on the visitor pass at the time of issuance. The expiration date could for example correspond to the visitor's allowed duration of stay in the foreign country.
```

File: docs/index.md
Comment (p2-apple) [inline docs/index.md:437]:
```suggestion
To address use cases such as the preceding one, an optional Health Card expiration date can be represented by the registered JWT `exp` claim (encoded as the number of seconds from 1970-01-01T00:00:00Z UTC, as specified by [RFC 7519](https://tools.ietf.org/html/rfc7519)). Verifiers are required to check the expiration, if present, and reject Health Cards with an `exp` value that is before the current verification date-time.
```

**Review by p2-apple — COMMENTED** [review]
Adding some suggestions to re-wording.

File: docs/index.md
Comment (pvillela) [reply docs/index.md:433]: @jmandel requested the inclusion of the note about expirations not being routinely used.

Comment (p2-apple) [inline docs/index.md:433]: Ah ok, then I would suggest adding something like:

```suggestion
SMART Health Cards contain factual information that is assured to be correct at the point of issuance and does not change with the passage of time. Therefore, **Health Cards generally do not expire** and an expiration date is not used. There are, however, situations where the ability to set an expiration date is beneficial.
```

Comment (pvillela) [reply docs/index.md:433]: I'm good with this rephrasing proposed by @p2-apple if @jmandel is OK with it.

File: docs/index.md
Comment (jmandel) [inline docs/index.md:435]:
```suggestion
One use case for issuing SMART Health Cards with an expiration date is a government entity issuing a vaccination card to foreign visitors for their use while in the destination country. This visitor's vaccination card is issued based on original documents presented by the visitor. Even with robust verification protocols, the government entity may not want to vouch for the validity of the visitor pass for an unlimited period of time. Importantly, the original document may be invalidated at some point in the future, e.g. by its signing keys being revoked. It may be impractical for the government entity issuing the visitor pass to track and reactively revoke the visitor pass. This risk can be mitigated by setting an expiration date on the visitor pass at the time of issuance. The expiration date could, for example, correspond to the visitor's allowed duration of stay in the foreign country.
```

File: docs/index.md
Comment (jmandel) [inline docs/index.md:437]:
```suggestion
To address use cases such as the preceding one, an optional SMART Health Card expiration date can be represented by the registered JWT `exp` claim (encoded as the number of seconds from 1970-01-01T00:00:00Z UTC, as specified by [RFC 7519](https://tools.ietf.org/html/rfc7519)). Verifiers are required to check the expiration, if present, and reject SMART Health Cards with an `exp` value that is before the current verification date-time.
```

**Comment by jmandel** [comment]
Thanks! I made a couple of tiny edits -- very happy with how this has shaped up.

**Comment by p2-apple** [comment]
> This impracticality applies generally to any large Issuer of health card data and is relevant to: #205

Sure. In this case it's compounded by the fact that the issuer has to monitor upstream certificate revocations (or whatever the upstream revocation story is) and then perform the downstream revocations.

**Comment by jmandel** [comment]
Makes sense to me @christianpaquin  re: example! I see this PR is tied to a branch in @pvillela so I'm not sure whether you have permission to make the change there (if not, we can handle it in a separate PR).

Will plan to merge this one tomorrow unless anyone raises a concern.

**Comment by christianpaquin** [comment]
The `pvillela:main` branch isn't up to date with the current head of the repository, so I can't add the `exp` to the new example-03 series. I don't think I have write access anyway, so I can do this on a separate PR.

**Comment by pvillela** [comment]
@christianpaquin, my fork's main is now up-to-date with your repo.

**Comment by christianpaquin** [comment]
I just added expiration value to the example-03 card.

**Review by jmandel — APPROVED** [review]

**Pull request merged**
